src/builder/edit_automod_rule.rs

```diff
@@ -1,133 +1,86 @@
-                let value = json!({
-                    "presets": presets,
-                });
-                self.0.insert("trigger_metadata", value);
-            },
-            _ => {},
-        }
-
+        self.trigger = Some(trigger);
         self
     }
 
     /// Set the actions which will execute when the rule is triggered.
-    pub fn actions<I>(&mut self, actions: I) -> &mut Self
-    where
-        I: IntoIterator<Item = Action>,
-    {
-        let actions = actions
-            .into_iter()
-            .map(|action| {
-                let kind = action.kind();
-                match action {
-                    Action::Alert(channel_id) => {
-                        json!({
-                            "type": kind,
-                            "metadata": {
-                                "channel_id": channel_id.0.to_string(),
-                            },
-                        })
-                    },
-                    Action::Timeout(duration) => {
-                        json!({
-                            "type": kind,
-                            "metadata": {
-                                "duration_seconds": duration,
-                            },
-                        })
-                    },
-                    Action::BlockMessage | Action::Unknown(_) => {
-                        json!({
-                            "type": kind,
-                        })
-                    },
-                }
-            })
-            .collect();
-
-        self.0.insert("actions", actions);
-
+    pub fn actions(&mut self, actions: Vec<Action>) -> &mut Self {
+        self.actions = Some(actions);
         self
     }
 
     /// Set whether the rule is enabled.
     pub fn enabled(&mut self, enabled: bool) -> &mut Self {
-        self.0.insert("enabled", Value::from(enabled));
-
+        self.enabled = Some(enabled);
         self
     }
 
     /// Set roles that should not be affected by the rule.
     ///
     /// Maximum of 20.
-    pub fn exempt_roles<I>(&mut self, roles: I) -> &mut Self
-    where
-        I: IntoIterator<Item = RoleId>,
-    {
-        let ids = roles.into_iter().map(|id| id.0.to_string()).collect();
-
-        self.0.insert("exempt_roles", ids);
-
+    pub fn exempt_roles(
+        &mut self,
+        roles: impl IntoIterator<Item = impl Into<RoleId>>,
+    ) -> &mut Self {
+        self.exempt_roles = Some(roles.into_iter().map(Into::into).collect());
         self
     }
 
     /// Set channels that should not be affected by the rule.
     ///
     /// Maximum of 50.
-    pub fn exempt_channels<I>(&mut self, channels: I) -> &mut Self
-    where
-        I: IntoIterator<Item = ChannelId>,
-    {
-        let ids = channels.into_iter().map(|id| id.0.to_string()).collect();
-
-        self.0.insert("exempt_channels", ids);
-
+    pub fn exempt_channels(&mut self, channels: Vec<ChannelId>) -> &mut Self {
+        self.exempt_channels = Some(channels);
```

Comment:
```suggestion
    pub fn exempt_channels(
        &mut self,
        channels: impl IntoIterator<Item = impl Into<ChannelId>>,
    ) -> &mut Self {
        self.exempt_channels = Some(channels.into_iter().map(Into::into).collect());
```

See above discussion.